packages/eslint-plugin/src/rules/ban-ts-comment.ts

```diff
@@ -214,14 +214,15 @@ export default createRule<[Options], MessageIds>({
                           : `/*${commentText}*/`,
                       );
                     },
+                    messageId: 'replaceTsIgnoreWithTsExpectError',
                   },
                 ],
               });
             } else {
               context.report({
                 data: { directive },
-                node: comment,
                 messageId: 'tsDirectiveComment',
+                node: comment,
```

Comment:
ideally `node`/`loc` would be first in the `context.report` as it's the most important piece of it...
IMO my ordering would be
1. `loc`
2. `node`
3. `messageId`
4. `data`
5. `fix`